pkg/tasks/prometheusadapter.go

```diff
@@ -154,3 +178,29 @@ func (t *PrometheusAdapterTask) Run() error {
 
 	return nil
 }
+
+func (t *PrometheusAdapterTask) deleteOldPrometheusAdapterAPIAuthenticationSecret(newName string) error {
+	deployment, err := t.client.KubernetesInterface().AppsV1beta2().Deployments(t.namespace).Get("prometheus-adapter", metav1.GetOptions{})
+
+	switch {
+	case apierrors.IsNotFound(err):
+		return nil
+	case err != nil:
+		return err
+	}
+
+	var name string
+	vs := deployment.Spec.Template.Spec.Volumes
+	for i := range vs {
+		if vs[i].Name == "api-auth" && vs[i].Secret != nil {
+			name = vs[i].Secret.SecretName
+			break
+		}
+	}
+
+	if name == "" || name == newName {
```

Comment:
I think this might be a bit too fragile, typically pruning logic like this is done by using label selectors. As in list all Secrets that have the label `name: api-auth`, and `not` the `hash: <latest-hash>`. That way even if for whatever reason we miss a rotation we still properly clean up in the future.

Reply:
agreed, I'll add this in a subsequent PR along with the serving certs.

Reply:
Sounds perfect.